### PR TITLE
Issue 1202 org rules from fes

### LINKED ISSUE #1202: Retrieve OrgRules from FES before trying backend
part of https://github.com/FlowCrypt/flowcrypt-android/issues/1298

### PR DESCRIPTION
**Pull request by DenBond7**
This PR added an ability to fetch OrgRules from FES

close #1202

----------------------------------

**Tests** _(delete all except exactly one)_:
- Tests added or updated

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [ ] addresses the issue it closes (if any)
- [ ] code is readable and understandable
- [ ] is accompanied with tests, or tests are not needed
- [ ] is free of vulnerabilities

### CODE REVIEW

File: FlowCrypt/src/main/java/com/flowcrypt/email/api/retrofit/FlowcryptApiRepository.kt [hunk 1]
```diff
@@ -46,14 +51,21 @@ class FlowcryptApiRepository : ApiRepository {
...
       val apiService = ApiHelper.getInstance(context).retrofit.create(ApiService::class.java)
       getResult(
         context = context,
         expectedResultClass = DomainOrgRulesResponse::class.java
-      ) { apiService.getDomainOrgRules(loginModel) }
+      ) {
+        if (fesUrl != null) {
+          apiService.getDomainOrgRules(fesUrl = fesUrl)
+        } else {
+          apiService.getDomainOrgRulesInternal(body = loginModel)
```
Comment (tomholub) [inline FlowCrypt/src/main/java/com/flowcrypt/email/api/retrofit/FlowcryptApiRepository.kt:66]: The distinction here is not clear enough. `getOrgRulesFromFes` and `getOrgRulesFromFlowCryptComBackend` would be clearer.

File: FlowCrypt/src/main/java/com/flowcrypt/email/api/retrofit/response/api/DomainOrgRulesResponse.kt [hunk 2]
```diff
@@ -24,7 +24,7 @@ import com.google.gson.annotations.SerializedName
 data class DomainOrgRulesResponse constructor(
   @SerializedName("error")
   @Expose override val apiError: ApiError? = null,
-  @SerializedName("domain_org_rules")
+  @SerializedName("domain_org_rules", alternate = ["clientConfiguration"])
```
Comment (tomholub) [inline FlowCrypt/src/main/java/com/flowcrypt/email/api/retrofit/response/api/DomainOrgRulesResponse.kt:27]: I recommend to create separate models. Backend and FES endpoints may look similar, but their stucture will often be different.

File: FlowCrypt/src/main/java/com/flowcrypt/email/util/GeneralUtil.kt [hunk 3]
```diff
@@ -79,6 +85,28 @@ class GeneralUtil {
       return activeNetwork?.isConnected ?: false
     }
 
+    suspend fun hasInternetAccess(): Boolean = withContext(Dispatchers.IO) {
+      val url = "https://www.google.com"
+      val connectionTimeoutInSeconds = 2000L
```
Comment (tomholub) [inline FlowCrypt/src/main/java/com/flowcrypt/email/util/GeneralUtil.kt:90]: `inMilliseconds`

**Review by tomholub — APPROVED** [review]
Code looks good. Will try the app

File: FlowCrypt/src/main/java/com/flowcrypt/email/util/GeneralUtil.kt (hunk 3 above)
Comment (DenBond7) [reply FlowCrypt/src/main/java/com/flowcrypt/email/util/GeneralUtil.kt:90]: :+1:

**Comment by tomholub** [comment]
I haven't tried it yet. It's ok - I'll test it from master.

**Comment by DenBond7** [comment]
Upps... Sorry.

For me approved PR means it's ready for merging.

**Comment by tomholub** [comment]
Understood - that's fair :-)

**Pull request merged**